Data/Aeson.hs

```diff
@@ -149,22 +149,22 @@ decodeStrict' = decodeStrictWith jsonEOF' fromJSON
 
 -- | Like 'decode' but returns an error message when decoding fails.
 eitherDecode :: (FromJSON a) => L.ByteString -> Either String a
-eitherDecode = eitherDecodeWith jsonEOF fromJSON
+eitherDecode = either (Left . uncurry formatError) Right . eitherDecodeWith jsonEOF fromJSON
```

Comment:
Two notes:

* These lines are longer than 80 characters.

* The expression `either (Left . uncurry formatError) Right . eitherDecodeWith x y` shows up repeatedly in this file as a result of your diff. That's a strong hint that you should extract it to a function and use that.